go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -803,17 +817,21 @@ func (npw *nodePortWatcher) AddEndpointSlice(epSlice *discovery.EndpointSlice) e
 
 	klog.V(5).Infof("Adding endpointslice %s in namespace %s", epSlice.Name, epSlice.Namespace)
 	nodeIPs := npw.nodeIPManager.ListAddresses()
-	hasLocalHostNetworkEp := hasLocalHostNetworkEndpoints([]*discovery.EndpointSlice{epSlice}, nodeIPs, svc)
-
 	epSlices, err := npw.watchFactory.GetEndpointSlices(svc.Namespace, svc.Name)
 	if err != nil {
-		klog.V(5).Infof("Could not fetch endpointslices for service %s/%s during endpointSliceAdd", svc.Name, svc.Namespace)
+		if !kerrors.IsNotFound(err) {
+			return fmt.Errorf("error retrieving endpointslices for service %s/%s during endpointslice add: %w",
+				svc.Namespace, svc.Name, err)
+		}
+		klog.V(5).Infof("No endpointslices found for service %s/%s during endpointslice add", svc.Name, svc.Namespace)
```

Comment:
we want to continue here even if epSlices is nil? I guess that was the past behaviour so we'll leave it like that...

Reply:
My point was to treat uniformly across the code these errors we get when retrieving resources and, possibly, distinguish between failed connections to the api server and not found resources when I saw that the code didn't do that, but it continued with the execution. 

At the same time, for instance here, I see now that it doesn't make much sense that we're adding an endpoint slice and then, when retrieving all endpoint slices for the service, we get 0 slices (that is, not even the one we wanted to add!) and we still continue with the execution. I'm more inclined to return the error regardless of the error type:
```
	epSlices, err := npw.watchFactory.GetEndpointSlices(svc.Namespace, svc.Name)
	if err != nil {
			return fmt.Errorf("error retrieving endpointslices for service %s/%s during endpointslice add: %w",
				svc.Namespace, svc.Name, err)
	}
```